tables/AlternateMans/UvwFile.h

```diff
@@ -173,6 +174,16 @@ class UvwFile {
             "Baselines are written in a non-ordered way: they need to be "
             "ordered either by antenna 1 or by antenna 2");
       }
+    } else {
+      // In the special case of a file with only auto-correlations, the
+      // positions of the antennas can not be established in this case, it's
+      // also not necessary, because a zero uvw can be returned for
+      // auto-correlations. However, it will cause StoreOrCheck() to be
+      // never called, and therefore the block is never written to the file, and
+      // upon read the nr. of rows can not be determined. So, if this is an
+      // auto-correlation that has not yet been written, mark the block as
+      // changed so it gets written.
+      block_is_changed_ = block_is_changed_ || !IsSet(antenna1);
```

Comment:
How is `!IsSet(antenna1)` an indicator for being an autocorrelation?

Reply:
This is in the `else` of `if (antenna1 != antenna2)`, hence this condition only runs on autocorrelations.

In theory we could use `block_is_changed_ = true` here, but then everytime a auto-correlation is set, it will cause rewrite of the whole block, which isn't necessary. So the `!IsSet(antenna1)` is a slight optimization, to only write if the position of the antenna is unknown, which makes sure that on a 'normal' MS, setting the auto-correlations again won't update the file needlessly.